akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -1143,6 +1143,14 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
       }
 
       publish(latestGossip)
+    } else {
+
+      val newGossip = localGossip.pruneTombstones(System.currentTimeMillis() - PruneGossipTombstonesAfter.toMillis)
```

Comment:
add a comment that tombstones may be reintroduced in a gossip merge but then they will be removed later again